src/layout/IHyprLayout.cpp

```diff
@@ -505,6 +574,75 @@ void IHyprLayout::onMouseMove(const Vector2D& mousePos) {
                 newPos = newPos + Vector2D((m_vBeginDragSizeXY - newSize).x, 0.0);
 
             CBox wb = {newPos, newSize};
+            if (*SNAPENABLED) {
+                if (*SNAPWINDOWGAP) {
+                    const int  gap  = *SNAPWINDOWGAP;
+                    const auto PID  = DRAGGINGWINDOW->getPID();
+                    const auto WSID = DRAGGINGWINDOW->workspaceID();
+
+                    for (auto& other : g_pCompositor->m_vWindows) {
```

Comment:
needs to ignore windows that:
 - aren't mapped
 - are fading out
 - are XWayland OR

Reply:
something like this?
```cpp
if (other->workspaceID() == WSID && other->getPID() != PID
 && other->m_bIsMapped
 && !other->m_bFadingOut
 && !other->m_bIsX11) {
```

Reply:
prefer guards.

```cpp
if (bad)
   continue;
```